compiler/test/dotty/tools/dotc/reporting/ErrorMessagesTests.scala

```diff
@@ -304,5 +304,29 @@ class ErrorMessagesTests extends ErrorMessagesTest {
       assertEquals("x", tree.name.show)
     }
 
+  @Test def superQualMustBeParent =
+    checkMessagesAfter("frontend") {
+      """
+        |class A {
+        |  def foo(): Unit = ()
+        |}
+        |
+        |class B {
+        |}
+        |
+        |class C extends A {
+        |  super[B].foo
+        |}
+      """.stripMargin
+    }
+      .expect { (ictx, messages) =>
+        implicit val ctx: Context = ictx
+        val defn = ictx.definitions
 
+        assertMessageCount(1, messages)
+        val SuperQualMustBeParent(qual, cls) :: Nil = messages
+
+        assert(qual.name.toString == "B", s"qualifier was: ${qual.name}")
+        assert(cls.denot.name.toString == "C", s"class was: ${cls.denot.name.toString}")
```

Comment:
I feel like this is a watered down version of what I should be testing (string comparison vs comparing the actual type object), but I'm not familiar enough with the compiler's type hierarchy. Suggestions welcome.